.github/workflows/nightly_dependency_tests.yaml

```
@@ -0,0 +1,76 @@
+name: Nightly dependencies@develop
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 23 * * *"
+
+concurrency:
+  # github.workflow: name of the workflow, so that we don't cancel other workflows
+  # github.event.pull_request.number || github.ref: pull request number or branch name if not a pull request
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  # Cancel in-progress runs when a new workflow with the same group name is triggered
+  # This avoids workflow runs on both pushes and PRs
+  cancel-in-progress: true
```

Since this won't run on pull requests or pushes but rather on schedule or manually, I will suggest using a different concurrency key to group the workflows (`github.ref` is available only for `pull_request:` and related events):

```suggestion
concurrency:
  # github.workflow: name of the workflow, so that we don't cancel other workflows
  # github.run_id || github.event_name: either the unique identifier for the job or the event that triggered it
  group: ${{ github.workflow }}-${{ github.run_id || github.event_name }}
  # Cancel in-progress runs when a new workflow with the same group name is triggered
  cancel-in-progress: true
```

I think it should work, because the scheduled events will be unaffected, and this should cancel duplicate `workflow_dispatch` runs. Otherwise, if it doesn't, we can drop the concurrency altogether later and cancel jobs manually.